pwndbg/gdblib/hooks.py

```diff
@@ -54,43 +46,28 @@ def on_stop() -> None:
     pwndbg.gdblib.strings.update_length()
 
 
-@pwndbg.gdblib.events.stop
-@pwndbg.gdblib.events.mem_changed
-@pwndbg.gdblib.events.reg_changed
-def memoize_on_stop() -> None:
-    reset_on_stop._reset()
-
-
-@pwndbg.gdblib.events.before_prompt
-def memoize_before_prompt() -> None:
-    reset_on_prompt._reset()
-
-
-@pwndbg.gdblib.events.cont
-def memoize_on_cont() -> None:
-    reset_on_cont._reset()
-
-
-@pwndbg.gdblib.events.new_objfile
-def memoize_on_new_objfile() -> None:
-    reset_on_objfile._reset()
-
-
-@pwndbg.gdblib.events.start
-def memoize_on_start() -> None:
-    while_running._start_caching()
-    reset_on_start._reset()
-
-
-@pwndbg.gdblib.events.exit
-def memoize_on_exit() -> None:
-    while_running._reset()
-    reset_on_exit._reset()
-
-
-@pwndbg.gdblib.events.thread
-def memoize_on_new_thread() -> None:
-    reset_on_thread._reset()
+import pwndbg.lib.cache
+
+pwndbg.lib.cache.connect_clear_caching_events(
+    (
```

Comment:
Not a big deal, but I think a `dict[str, Tuple]` would be a little nicer here instead of the nested tuples.